File: 1251-1500/1351. Count Negative Numbers in a Sorted Matrix/Solution.java
class Solution {
    public int countNegatives(int[][] grid) {
        int n = grid.length ; 
        int m = grid[0].length ;
        int ans = 0 ; 
        int col = m ; 
        for( int i=0 ; i<n ; ++i ){
            while( col > 0 && grid[i][col-1] < 0 ){
                col-- ; 
            }
            ans += m - col ; 
        }
        return ans ;  
    }
}

File: 1251-1500/1351. Count Negative Numbers in a Sorted Matrix/solution.py
from typing import List

class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        n = len(grid)
        m = len(grid[0])
        ans = 0
        col = m
        for i in range(n):
            while col > 0 and grid[i][col - 1] < 0:
                col -= 1
            ans += m - col

        return ans
